src/components/avatar/index.js

```diff
@@ -122,6 +121,10 @@ export default class Avatar extends BaseComponent {
      * ribbon label custom style
      */
     ribbonLabelStyle: Text.propTypes.style,
+    /**
+     * custom ribbon
+     */
+    customRibbon: PropTypes.oneOfType([PropTypes.object, PropTypes.func, PropTypes.element]),
```

Comment:
you need to decide on the type of `customRibbon`. 
first of all, `object` is irrelevant, you can render a plain object. 
second, there's a difference between `element` and `func`. 
if you're going with element then you just place it in your render `{customRibbon}`
if you're going with a function you actual need to invoke it - `{customRibbon()}`

Usually going with `func` type is good if you want to pass things to the function (for example the `ribbonLabel`)